sql/table.cc

```diff
@@ -7057,18 +7057,21 @@ int TABLE::update_default_fields()
   DBUG_RETURN(res);
 }
 
-bool TABLE::vers_update_fields()
+void TABLE::vers_update_fields()
 {
   DBUG_ENTER("vers_update_fields");
 
   DBUG_ASSERT(versioned());
-  DBUG_ASSERT(!vers_start_field()->set_time());
-  DBUG_ASSERT(!vers_end_field()->set_max_timestamp());
+  bool res= !vers_start_field()->set_time();
+  DBUG_ASSERT(res);
```

Comment:
I also use something like:

    if (vers_start_field()->set_time())
         DBUG_ASSERT(0);

It doesn't need additional variable.